run.nextflow.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+import time
+
+# DIR where the current script resides
+DIR = os.path.dirname(os.path.abspath(__file__))
+
+# set default values for arguments if not provided
+pipeline = sys.argv[1] if len(sys.argv) > 1 else 'nanopore'
+profile = sys.argv[2] if len(sys.argv) > 2 else 'gls'
+root_dir = sys.argv[3] if len(sys.argv) > 3 else 'gs://prj-int-dev-covid19-nf-gls'
+snapshot_date = sys.argv[4] if len(sys.argv) > 4 else '2023-03-15'
+concurrency = sys.argv[5] if len(sys.argv) > 5 else '100'
+batch_size = sys.argv[6] if len(sys.argv) > 6 else '15000'
+dataset_name = sys.argv[7] if len(sys.argv) > 7 else 'sarscov2_metadata'
+project_id = sys.argv[8] if len(sys.argv) > 8 else 'prj-int-dev-covid19-nf-gls'
+test_submission = 'false'
+
+# Row count and batches
+table_name = f"{pipeline}_to_be_processed"
+sql = f"SELECT count(*) AS total FROM {project_id}.{dataset_name}.{table_name}"
+row_count = subprocess.check_output(f"bq --project_id={project_id} --format=csv query --use_legacy_sql=false \"{sql}\" | grep -v total", shell=True).decode('utf-8')

```

Comment:
Big Query has native Python APIs. You might want to consider replace the subprocess calls with the native Python APIs instead.